var/spack/repos/builtin/packages/qtgraph/package.py

```diff
@@ -69,11 +70,22 @@ def setup_environment(self, spack_env, run_env):
         spack_env.set('GRAPHVIZ_ROOT', self.spec['graphviz'].prefix)
         spack_env.set('INSTALL_ROOT', self.prefix)
 
+        # NOTE: Spack runs setup_environment twice, once pre-build to set up
+        # the build environment, and once post-installation to determine
+        # the environment variables needed at run-time to add to the module
+        # file. The script we need to source is only present post-installation,
+        # so check for its existence before sourcing.
+        # TODO: At some point we should split setup_environment into
+        # setup_build_environment and setup_run_environment to get around
+        # this problem.
+
         # The implementor has set up the library and include paths in
         # a non-conventional way.  We reflect that here.
-        run_env.prepend_path(
-            'LD_LIBRARY_PATH', join_path(
-                self.prefix.lib64,
-                '{0}'.format(self.spec['qt'].version.up_to(3))))
+        qtgraphlibs = join_path(
+            self.spec['qtgraph'].libs,
```

Comment:
Libs should return a list of library files, not the lib or lib64 directory. Take a look at the `LibraryList` docs.

Reply:
I forget, does libs raise an error if it can't find anything or does it just return None/an empty list? You shouldn't need to check if the path exists, it can't return a non-existing path. Just check if it's empty.